### PR TITLE
Added in local context to TypeResolutionEnvironment

### PR DESCRIPTION
**Pull request by bbakerman**
And also cleaned up TypeResolutionEnvironment.java / TypeResolutionParameters.java a bit since there was lots of repeated code and really they should be merged at some point

See https://github.com/graphql-java/graphql-java/issues/2693

### CODE REVIEW

File: src/main/java/graphql/TypeResolutionEnvironment.java [hunk 1]
```diff
@@ -24,24 +27,20 @@
     private final GraphQLSchema schema;
     private final Object context;
     private final GraphQLContext graphQLContext;
+    private final Object localContext;
     private final DataFetchingFieldSelectionSet fieldSelectionSet;
 
-    public TypeResolutionEnvironment(Object object,
```
Comment (bbakerman) [inline src/main/java/graphql/TypeResolutionEnvironment.java:29]: Yes this is breaking API - but this should not have been made public - a builder pattern (which we started AFTER this class was invented) would have not made this a breaking change.

We really dont want consumers making these

Comment (bbakerman) [reply src/main/java/graphql/TypeResolutionEnvironment.java:29]: we could got the whole hog and merge the two classes and put the builder int this class.  I chose not to do this - progress over perfection

File: src/main/java/graphql/execution/ResolveType.java [hunk 2]
```diff
@@ -20,33 +20,24 @@
...
+                .fieldType(fieldType)
+                .value(source)
+                .argumentValues(executionStepInfo.getArguments())
+                .selectionSet(fieldSelectionSet)
+                .context(executionContext.getContext())
+                .graphQLContext(executionContext.getGraphQLContext())
+                .localContext(localContext)
+                .schema(executionContext.getGraphQLSchema())
+                .build();
         if (fieldType instanceof GraphQLInterfaceType) {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ResolveType.java:37]: Simplified - there was never a need for the union vs interface type - there are never treated special outside this spot and again only for resolver lookup

File: src/test/groovy/graphql/TypeResolverExecutionTest.groovy [hunk 3]
```diff
@@ -580,6 +582,82 @@ class TypeResolverExecutionTest extends Specification {
...
+
+        TypeResolver typeResolver = new TypeResolver() {
+            @Override
+            GraphQLObjectType getType(TypeResolutionEnvironment env) {
+                assert env.getField().getName() == "foo"
+                assert env.getFieldType() == env.getSchema().getType("BarInterface")
+                assert env.getContext() == "Context"
+                assert env.getGraphQLContext().get("x") == "graphqlContext"
+                assert env.getLocalContext() == "LocalContext"
+                return env.getSchema().getObjectType("NewBar")
```
Comment (bbakerman) [inline src/test/groovy/graphql/TypeResolverExecutionTest.groovy:618]: we never had a full integration test that ensured that these values get set at runtime

**Review by andimarek — APPROVED** [review]

**Pull request merged**